db/db_impl/db_impl_open.cc

```diff
@@ -1203,13 +1203,18 @@ Status DBImpl::ProcessLogFiles(
   PredecessorWALInfo predecessor_wal_info;
 
   for (auto wal_number : wal_numbers) {
```

Comment:
nit: would you mind early return on this loop on non-ok status ... probably my bad of not doing that already in `status = ProcessLogFile()` ....